### ISSUE TITLE
`dvipng` not being found during github build

### ISSUE THREAD

**Issue by mkenworthy** [showyourwork/showyourwork#377]
My paper builds on my computer, but fails when pushed to github. I have figures that have latex in them, so I have put in `tinytex` into the build process. Based on previous conversations, my `build.yml` and `build-pull-request.yml` have:

 ```
     - name: Install TinyTex for matplotlib LaTeX rendering
        id: tinytex
        shell: bash -l {0}
        run: |
          wget -qO- "https://yihui.org/tinytex/install-bin-unix.sh" | sh
          sudo ~/bin/tlmgr install type1cm cm-super underscore dvipng

```

...and I've added the `PATH` into my plotting scripts.

However, `dvipng` does not seem to be found during the build process on gihub, I get an error of

```
  File "/home/runner/work/yses2b_orbit/yses2b_orbit/.snakemake/conda/d3e94cc67be7e36e11d5525cc948392a_/lib/python3.11/site-packages/matplotlib/texmanager.py", line 259, in _run_checked_subprocess
    raise RuntimeError(
RuntimeError: Failed to process string with tex because dvipng could not be found
```

...any ideas as to what's going on?

**Comment by dfm** [comment]
I'm sorry for the delayed response! Unfortunately I'm not too sure what the problem is here, or if you're still running into issues. Please let us know if you found a solution, or we can try to iterate on it now that I'm back from leave.

**Comment by HealthyPear** [comment]
Hi @mkenworthy ,

I am going through old issues to scoop that ones that could be closed :)

I think this might be one of them: if your aim is to produce vector graphics into the resulting manuscript there are options easier than installing a TeX engine on top of tectonic (the one that showyourwork uses behind the scenes), e.g.

- matplotlib has a PGF plots based backend

```python
import matplotlib as mpl
mpl.use("pgf")
mpl.rcParams.update({
    "pgf.texsystem": "pdflatex",  # or "xelatex"/"lualatex"
    "text.usetex": True,
    "font.family": "serif",
})

import matplotlib.pyplot as plt
plt.plot([1, 2, 3], [1, 4, 9])
plt.xlabel(r"$x$")
plt.ylabel(r"$y$")
plt.savefig("figure.pgf")
```

then in the TeX file

```tex
\usepackage{pgf}

\input{figure.pgf}
```



- you can use the currently unmaintained but still useful https://github.com/nschloe/tikzplotlib
- this cool thing I just found out https://github.com/gpoore/pythontex

Does this help you?

**Comment by mkenworthy** [comment]
Many thanks for coming back to this - this was me reverse engineering how to get matplotlib to use the tex engine for labeling figures, which didn't work without matplotlib's access to a separate latex install, IIRC.

**Comment by HealthyPear** [comment]
Yeah, I do not think there is a straightforward way to make matplotlib talk to tectonic (but I can ask to make sure), so my suggestion for now is to create the entire plot in latex using something else, and then let tectonic process it without "knowing" that it's a plot.

**Comment by HealthyPear** [comment]
This should put a (temporary) nail in the coffin :/

https://github.com/matplotlib/matplotlib/issues/22715

**Comment by HealthyPear** [comment]
> Many thanks for coming back to this - this was me reverse engineering how to get matplotlib to use the tex engine for labeling figures, which didn't work without matplotlib's access to a separate latex install, IIRC.

sorry, coming back to this I might have overcomplicated things in my previous responses above

isn't this the use case that you describe (without having to install tinytex)?
https://show-your.work/en/latest/faqs/#using-latex-fonts-in-matplotlib-without-installing-latex

**Comment by mkenworthy** [comment]
I think I eventually worked around not using LaTeX in my figures, so avoided the issue. I'll use your suggested method when I encounter it again.
